Fixed fallocate inode operation return value type.

Signed-Off-By: Mingming Cao <cmm@us.ibm.com>
Index: linux-2.6.22-rc1/include/linux/fs.h
===================================================================
--- linux-2.6.22-rc1.orig/include/linux/fs.h	2007-05-15 17:38:59.000000000 -0700
+++ linux-2.6.22-rc1/include/linux/fs.h	2007-05-15 17:39:20.000000000 -0700
@@ -1148,7 +1148,7 @@
 	ssize_t (*listxattr) (struct dentry *, char *, size_t);
 	int (*removexattr) (struct dentry *, const char *);
 	void (*truncate_range)(struct inode *, loff_t, loff_t);
-	long (*fallocate)(struct inode *inode, int mode, loff_t offset,
+	int (*fallocate)(struct inode *inode, int mode, loff_t offset,
 			  loff_t len);
 };
 
