Remove unnecessary exported symbols.

Signed-Off-By: Mingming Cao <cmm@us.ibm.com>

Index: linux-2.6.22-rc1/fs/ext4/extents.c
===================================================================
--- linux-2.6.22-rc1.orig/fs/ext4/extents.c	2007-05-15 16:49:04.000000000 -0700
+++ linux-2.6.22-rc1/fs/ext4/extents.c	2007-05-15 16:49:32.000000000 -0700
@@ -2572,11 +2572,3 @@
 
 	return ret > 0 ? ret2 : ret;
 }
-
-EXPORT_SYMBOL(ext4_mark_inode_dirty);
-EXPORT_SYMBOL(ext4_ext_invalidate_cache);
-EXPORT_SYMBOL(ext4_ext_insert_extent);
-EXPORT_SYMBOL(ext4_ext_walk_space);
-EXPORT_SYMBOL(ext4_ext_find_goal);
-EXPORT_SYMBOL(ext4_ext_calc_credits_for_insert);
-
