Remove unnecessary exported symbols.

Signed-Off-By: Mingming Cao <cmm@us.ibm.com>

Index: linux-2.6.22-rc1/fs/ext4/extents.c
===================================================================
diff --git a/fs/ext4/extents.c b/fs/ext4/extents.c
index ffe7cb6..1630317 100644
--- a/fs/ext4/extents.c
+++ b/fs/ext4/extents.c
@@ -2157,11 +2157,3 @@ int ext4_ext_writepage_trans_blocks(struct inode *inode, int num)
 
 	return needed;
 }
-
-EXPORT_SYMBOL(ext4_mark_inode_dirty);
-EXPORT_SYMBOL(ext4_ext_invalidate_cache);
-EXPORT_SYMBOL(ext4_ext_insert_extent);
-EXPORT_SYMBOL(ext4_ext_walk_space);
-EXPORT_SYMBOL(ext4_ext_find_goal);
-EXPORT_SYMBOL(ext4_ext_calc_credits_for_insert);
-
