From: Mingming Cao <cmm@us.ibm.com>

Fix ext4 to reference CONFIG_JBD2_DEBUG

Signed-off-by: Mingming Cao <cmm@us.ibm.com>

Index: linux-2.6.22/fs/ext4/balloc.c
===================================================================
--- linux-2.6.22.orig/fs/ext4/balloc.c	2007-07-08 16:32:17.000000000 -0700
+++ linux-2.6.22/fs/ext4/balloc.c	2007-07-09 15:25:23.000000000 -0700
@@ -517,7 +517,7 @@ do_more:
 		/*
 		 * An HJ special.  This is expensive...
 		 */
-#ifdef CONFIG_JBD_DEBUG
+#ifdef CONFIG_JBD2_DEBUG
 		jbd_unlock_bh_state(bitmap_bh);
 		{
 			struct buffer_head *debug_bh;
@@ -1597,7 +1597,7 @@ allocated:
 
 	performed_allocation = 1;
 
-#ifdef CONFIG_JBD_DEBUG
+#ifdef CONFIG_JBD2_DEBUG
 	{
 		struct buffer_head *debug_bh;
 
Index: linux-2.6.22/fs/ext4/ioctl.c
===================================================================
--- linux-2.6.22.orig/fs/ext4/ioctl.c	2007-07-09 15:25:08.000000000 -0700
+++ linux-2.6.22/fs/ext4/ioctl.c	2007-07-09 15:25:23.000000000 -0700
@@ -141,7 +141,7 @@ flags_err:
 		ext4_journal_stop(handle);
 		return err;
 	}
-#ifdef CONFIG_JBD_DEBUG
+#ifdef CONFIG_JBD2_DEBUG
 	case EXT4_IOC_WAIT_FOR_READONLY:
 		/*
 		 * This is racy - by the time we're woken up and running,
@@ -283,7 +283,7 @@ long ext4_compat_ioctl(struct file *file
 	case EXT4_IOC32_SETVERSION_OLD:
 		cmd = EXT4_IOC_SETVERSION_OLD;
 		break;
-#ifdef CONFIG_JBD_DEBUG
+#ifdef CONFIG_JBD2_DEBUG
 	case EXT4_IOC32_WAIT_FOR_READONLY:
 		cmd = EXT4_IOC_WAIT_FOR_READONLY;
 		break;
