Add a mount option to turn off delayed allocation.

Signed-off-by: Mingming Cao <cmm@us.ibm.com>
---
Index: linux-2.6.22-rc5/fs/ext4/super.c
===================================================================
--- linux-2.6.22-rc5.orig/fs/ext4/super.c	2007-06-20 17:14:19.000000000 -0700
+++ linux-2.6.22-rc5/fs/ext4/super.c	2007-06-20 17:15:24.000000000 -0700
@@ -737,7 +737,7 @@ enum {
 	Opt_jqfmt_vfsold, Opt_jqfmt_vfsv0, Opt_quota, Opt_noquota,
 	Opt_ignore, Opt_barrier, Opt_err, Opt_resize, Opt_usrquota,
 	Opt_grpquota, Opt_extents, Opt_noextents, Opt_noversion,
-	Opt_delayed_alloc,
+	Opt_delayed_alloc, Opt_nodelayed_alloc,
 };
 
 static match_table_t tokens = {
@@ -793,6 +793,7 @@ static match_table_t tokens = {
 	{Opt_noextents, "noextents"},
 	{Opt_noversion, "noversion"},
 	{Opt_delayed_alloc, "delalloc"},
+	{Opt_nodelayed_alloc, "nodelalloc"},
 	{Opt_err, NULL},
 	{Opt_resize, "resize"},
 };
@@ -1117,6 +1118,9 @@ clear_qf_name:
 		case Opt_delayed_alloc:
 			set_opt(sbi->s_mount_opt, DELAYED_ALLOC);
 			break;
+		case Opt_nodelayed_alloc:
+			clear_opt(sbi->s_mount_opt, DELAYED_ALLOC);
+			break;
 		case Opt_ignore:
 			break;
 		case Opt_resize:
