
Signed-off-by: Mingming Cao <cmm@us.ibm.com>
---
 fs/quota_v2.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.28-rc2/fs/quota_v2.c
===================================================================
--- linux-2.6.28-rc2.orig/fs/quota_v2.c	2008-10-31 14:19:00.000000000 -0700
+++ linux-2.6.28-rc2/fs/quota_v2.c	2008-10-31 14:19:03.000000000 -0700
@@ -116,8 +116,8 @@ static int v2_read_file_info(struct supe
 		info->dqi_maxblimit = 0xffffffff;
 		info->dqi_maxilimit = 0xffffffff;
 	} else {
-		info->dqi_maxblimit = 0x7fffffffffffffff;
-		info->dqi_maxilimit = 0x7fffffffffffffff;
+		info->dqi_maxblimit = 0x7fffffffffffffffULL;
+		info->dqi_maxilimit = 0x7fffffffffffffffULL;
 	}
 	info->dqi_bgrace = le32_to_cpu(dinfo.dqi_bgrace);
 	info->dqi_igrace = le32_to_cpu(dinfo.dqi_igrace);
