diff -uprN linux-2.6.22-rc5/fs/ext4/group.h linux-2.6.22-rc5-patch/fs/ext4/group.h
--- linux-2.6.22-rc5/fs/ext4/group.h	2007-06-21 16:32:52.000000000 -0700
+++ linux-2.6.22-rc5-patch/fs/ext4/group.h	2007-06-21 16:30:33.000000000 -0700
@@ -8,7 +8,7 @@

 #ifndef _LINUX_EXT4_GROUP_H
 #define _LINUX_EXT4_GROUP_H
-#if defined(CONFIG_CRC16) || defined(CONFIG_CRC16_MODULE)
+#if defined(CONFIG_CRC16)
 #include <linux/crc16.h>
 #endif

diff -uprN linux-2.6.22-rc5/fs/ext4/super.c linux-2.6.22-rc5-patch/fs/ext4/super.c
--- linux-2.6.22-rc5/fs/ext4/super.c	2007-06-21 16:32:53.000000000 -0700
+++ linux-2.6.22-rc5-patch/fs/ext4/super.c	2007-06-21 16:30:22.000000000 -0700
@@ -1232,7 +1232,7 @@ static int ext4_setup_super(struct super
 	return res;
 }

-#if !defined(CONFIG_CRC16) && !defined(CONFIG_CRC16_MODULE)
+#if !defined(CONFIG_CRC16)
 /** CRC table for the CRC-16. The poly is 0x8005 (x16 + x15 + x2 + 1) */
 __u16 const crc16_table[256] = {
 	0x0000, 0xC0C1, 0xC181, 0x0140, 0xC301, 0x03C0, 0x0280, 0xC241,
